### PR TITLE
GS-13 Chore write pseudo code

### PR DESCRIPTION
**Pull request by murpjack**

### CODE REVIEW

File: src/lib/state.rs [hunk 1]
```diff
@@ -1,17 +1,55 @@
...
+use serde::{Deserialize, Serialize};
 use zbus::dbus_interface;
-use crate::transaction::{Transaction, Currency, Frequency};
 
-#[derive(Debug)]
+#[derive(Debug, Serialize, Deserialize)]
 pub struct State {
-    pub transactions: Vec<Transaction>,
+    pub incomes: Vec<Transaction>,
+    pub outgoings: Vec<Transaction>,
```
Comment (miltonio94) [inline src/lib/state.rs:11]: Would it be worth making a OutgoingTransaction type and an incomingTransaction type so we could deferentia between the two on a type level

File: src/lib/error.rs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+use std::fmt::Debug;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Failed to parse json.")]
```
Comment (miltonio94) [inline src/lib/error.rs:5]: This is really cool! I didn't know it was possible

File: src/lib/state.rs [hunk 3]
```diff
@@ -33,13 +71,13 @@ impl State {
...
-        self.transactions.push(transaction);
+        self.outgoings.push(transaction);
 
         format!("Transactions: {:?}", self)
     }
 
     #[dbus_interface(property)]
-    async fn transactions(&self) -> String {
-        format!("{:?}", self.transactions)
+    async fn outgoings(&self) -> String {
```
Comment (miltonio94) [inline src/lib/state.rs:80]: Maybe we should serialize this into json so it could be easy to deserialise this on the other side

**Review by miltonio94 — APPROVED** [review]
Looks good to me. I have noted two questions for your consideration

File: src/lib/state.rs (hunk 1 above)
Comment (murpjack) [reply src/lib/state.rs:11]: Possibly, though this seems an very trivial thing to implement if the type of transactions in and out diverge. I would request we leave for no and revisit.

File: src/lib/state.rs (hunk 3 above)
Comment (murpjack) [reply src/lib/state.rs:80]: Are you saying the type could stay the same but the String out would be a JSON string?
I would be in favour of this. If so, I can implement relatively simply now that a Transaction type has Serde Traits derived

File: src/lib/state.rs (hunk 1 above)
Comment (miltonio94) [reply src/lib/state.rs:11]: Yeah, sounds good to me

File: src/lib/state.rs (hunk 3 above)
Comment (miltonio94) [reply src/lib/state.rs:80]: Yup, the type stays the same we just returned a stringyfied JSON

**Pull request merged**